Add browser support

riknoll opened the pull request:
This PR splits pxt-mkc into three packages:

1. mkc-core, which contains the shared code for the other two packages and has no node dependencies
2. mkc-node, which contains the node CLI
3. mkc-browser, which contains a browser implementation of the language service

mkc-node will be published as the existing makecode package. The other two are new packages. The CLI should work exactly the same as before.

Opening as a draft PR because converting this repo into a monorepo broke a few things (e.g. the build, semantic-release)

pelikhan commented on packages/mkc-browser/package.json:
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "mkc-browser",
```
there is already an unrealated "mkc" package in npm, so we should stay away from this name

pelikhan replied:
https://www.npmjs.com/package/mkc

pelikhan commented on packages/mkc-browser/package.json:
```diff
@@ -0,0 +1,70 @@
+  },
+  "author": "",
+  "license": "MIT",
+  "homepage": "https://makecode.com/docs",
```
probably point to something more precise

pelikhan commented on packages/mkc-browser/package.json:
```diff
@@ -0,0 +1,70 @@
+  ],
+  "preferGlobal": false,
+  "engines": {
+    "node": ">= 11.0.0"
```
should we bump this to 14 at least?

pelikhan commented on packages/mkc/package.json:
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "mkc-core",
```
same as mkc-browser

pelikhan commented:
Can you update the local development section of the README with information about various package; and/or update build instructions if needed.

mmoskal commented on .gitignore:
```diff
@@ -5,3 +5,5 @@ node_modules
 *.vsix
 src/simloaderfiles.ts
+packages/mkc/src/simloaderfiles.ts
+packages/mkc-browser/src/workerFiles.ts
```
makecode-browser not mkc-browser?

mmoskal commented on packages/makecode-browser/src/workerFiles.ts:
```diff
@@ -0,0 +1,178 @@
+export const workerJs = `
```
was this ignored?

riknoll commented:
alright @pelikhan, I updated the packages to all be `makecode-*` instead of mkc and fixed the build/updated the readme. I also added semantic-release-monorepo which *should* fix semantic-release though I obviously haven't tested that yet.

Take another look real quick?

mmoskal commented on packages/makecode-core/src/simloaderfiles.ts:
```diff
@@ -0,0 +1,357 @@
+export const simloaderFiles: Record<string, string> = {
```
should be ignored?

mmoskal reviewed (approved):
looks good!
